fix(Profile/ContactsView): updated contacts layout

alexandraB99 opened the pull request:
Closes #6262

### What does the PR do
 All contacts should be shown in one long list, should the list is longer than the page, the page should scroll down

### Affected areas
ContactsView

### Screenshot of functionality (including design for comparison)
https://user-images.githubusercontent.com/31625338/189392161-7c795d4b-e851-48c3-b449-334505eb33a2.mov

https://user-images.githubusercontent.com/31625338/189392205-85d58715-62f2-4e6e-98b5-4208bbd0512c.mov


***The black images is list of dummy contacts

status-im-auto commented:
### Jenkins Builds
| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 044a2700 | [#1](https://ci.status.im/job/status-desktop/job/branches/job/macos/job/PR-7080/1/) | 2022-08-18 14:53:36 | ~10 min | `macos` | [:package:`dmg`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Desktop-220818-144331-044a27-pr7080.dmg) |

osmaczko commented on ui/app/AppLayouts/Profile/views/ContactsView.qml:
```diff
@@ -117,10 +117,10 @@ SettingsContentBase {
-                            (stackLayout.height-mutualContacts.contactsListHeight))
-                    scrollbarOn: mutualContacts.contactsListHeight > (stackLayout.height/2) ?
-                                 (contactsListHeight > (stackLayout.height/2)) : (contactsListHeight > parent.height)
+                    height: ((contactsListHeight < (root.height/2)) ? contactsListHeight :
+                                                                      (root.height-mutualContacts.contactsListHeight))
```
:eyes: Conditions like this is qml code smell..

Could you please try to use `ColumnLayout` instead of `Column` for items positioning and height management? `Layout.fillHeight: true` should do the trick.

osmaczko commented on ui/app/AppLayouts/Profile/views/ContactsView.qml:
```diff
@@ -117,10 +117,10 @@ SettingsContentBase {
-                                 (contactsListHeight > (stackLayout.height/2)) : (contactsListHeight > parent.height)
+                    height: ((contactsListHeight < (root.height/2)) ? contactsListHeight :
+                                                                      (root.height-mutualContacts.contactsListHeight))
+                    scrollbarOn: mutualContacts.contactsListHeight > (root.height/2) ?
+                                     (contactsListHeight > (root.height/2)) : (contactsListHeight > parent.height)
```
It should simply be:
```
scrollbarOn: contentHeight > height
```

osmaczko commented on ui/app/AppLayouts/Profile/views/ContactsView.qml:
```diff
@@ -138,8 +138,8 @@ SettingsContentBase {
                     height: (contactsListHeight+50)
-                    scrollbarOn: verifiedContacts.contactsListHeight > (stackLayout.height/2) ?
-                                 (contactsListHeight > (stackLayout.height/2)) : (contactsListHeight > parent.height)
+                    scrollbarOn: verifiedContacts.contactsListHeight > (root.height/2) ?
+                                     (contactsListHeight > (root.height/2)) : (contactsListHeight > parent.height)
```
Same as above.

osmaczko commented on ui/app/AppLayouts/Profile/views/ContactsView.qml:
```diff
@@ -169,11 +169,11 @@ SettingsContentBase {
-                                 (contactsListHeight > (stackLayout.height - sentRequests.contactsListHeight))
+                    height: ((contactsListHeight < (root.height/2)) ? contactsListHeight :
+                                                                      (root.height-sentRequests.contactsListHeight))
+                    scrollbarOn: (sentRequests.contactsListHeight > root.height) ?
+                                     (contactsListHeight > (root.height/2)) :
+                                     (contactsListHeight > (root.height - sentRequests.contactsListHeight))
```
Same as above.

osmaczko commented on ui/app/AppLayouts/Profile/views/ContactsView.qml:
```diff
@@ -213,9 +213,9 @@ SettingsContentBase {
-                    scrollbarOn: (receivedRequests.contactsListHeight > (stackLayout.height/2)) ?
-                                 (contactsListHeight > (stackLayout.height/2)) :
-                                 (contactsListHeight > (stackLayout.height - receivedRequests.contactsListHeight))
+                    scrollbarOn: (receivedRequests.contactsListHeight > (root.height/2)) ?
+                                     (contactsListHeight > (root.height/2)) :
+                                     (contactsListHeight > (root.height - receivedRequests.contactsListHeight))
```
Same as above.

glitchminer commented:
## Issues found

https://user-images.githubusercontent.com/50769329/188175443-1b4ab53b-06d7-4451-bae0-8047752d44cd.mp4

---
### 1. OPEN Trackpad cannot control main scrollbar
Actual - see video above

---
### 2. OPEN Mouse cannot control scrollbar for individual verified and unverified contact sections
Actual - see video above

---
### 3. OPEN After searching scrollbars remain present for individual sections
![image](https://user-images.githubusercontent.com/50769329/188169132-c2abcb81-9d3f-4f03-9177-868feaa53a71.png)

---
### 4. OPEN [OBSERVATION] With enough vertical height in the app all contacts are shown and scrollbar still appears

alexandraB99 commented:
@glitchminer good findings! @John-44 this is an issue for a while now, we need to somehow indicate the user that there are more contacts so they need to scroll and the idea was to keep the contact's list scroll bar always visible when there is more contacts to scroll. Apparently there is also the main scrolling behavior of the overall page which makes it a bit weird, also with overlapping of the scroll bars not being able to control the contact's list scrollbar. Other than this, there is no clear separation between the lists, how could we improve a bit the UI here?

endulab commented on ui/app/AppLayouts/Profile/panels/ContactsListPanel.qml:
```diff
@@ -21,8 +21,8 @@ Item {
     property int panelUsage: Constants.contactsPanelUsage.unknownPosition
-    property bool scrollbarOn: false
-    readonly property int contactsListHeight: ((contactsList.count * contactsList.itemAtIndex(0).implicitHeight)+title.height)
+    property bool scrollbarOn: ((contactsListHeight - 50) > (root.height-50))
```
@alexandraB99 this magic "50" shows up in a few files. Maybe it makes sense to have a constant for it somewhere, eg. in Theme ?

alexandraB99 replied:
not really, it's really specific to this contact list, a few pixels margin so that it can be scrolled completely at the bottom and the last item to be fully visible. I can let's say create a new variable in ContactsListPanel to keep it there, but maybe it's too much in this case.

PascalPrecht commented:
>  we need to somehow indicate the user that there are more contacts so they need to scroll and the idea was to keep the contact's list scroll bar always visible when there is more contacts to scroll.

@John-44 @alexandraB99 So this particular issue came up in the past a few times as we run into this in many different places. Back then I proposed to introduce some sort of inset dropshadow in scrollable areas that would indicate that there's more content to scroll through. @benjthayer has looked into this as well as part of the discord import tool. Maybe it's time to revisit this as a solution?

John-44 commented:
@alexandraB99 first of all there should be only a single scrollable area in the screen body area, not sperate scrollable areas for 'Identity verified contacts' and 'contacts'.   This should be a single list, with two sections ('Identity verified contacts' and 'contacts') that scrolls as one long list.

Secondly, if there are more contacts than can fit inside the available vertical window space, this list of contacts should extend all the way to the bottom of the window, where it gets cut off.  In most (but not all) cases the user should see the last contact being cutoff by the window, this will indicate to the user that there are more contacts the user can scroll to.  We will come up with something nicer when we do the desktop redesign post-MVP, but for now this should be good enough.

alexandraB99 commented:
@glitchminer updated based on the comment above, please have a second look!

glitchminer commented:
> @glitchminer updated based on the comment above, please have a second look!

Hi @alexandraB99, closed off the other issues as superseded and added issue 6 https://github.com/status-im/status-desktop/pull/7080#issuecomment-1235596212

alexandraB99 commented:
> > @glitchminer updated based on the comment above, please have a second look!
> 
> Hi @alexandraB99, closed off the other issues as superseded and added issue 6 [#7080 (comment)](https://github.com/status-im/status-desktop/pull/7080#issuecomment-1235596212)

@glitchminer thanks! I've updated it, please re-check and let me know if anything :)

glitchminer commented:
Retested, same issue present.

alexandraB99 commented:
> Retested, same issue present.

@glitchminer thanks for retesting, PR updated, issue should be fixed now!

glitchminer commented:
@alexandraB99, I'm still seeing the same thing
![image](https://user-images.githubusercontent.com/50769329/190386251-8a3a2a92-e75e-43fc-af27-1f4af59532ea.png)

glitchminer reviewed (approved):
Tested and approved.
